pyvelox/pyvelox.h

```diff
@@ -419,9 +516,58 @@ inline void addVectorBindings(
           })
       .def("encoding", &BaseVector::encoding)
       .def("append", [](VectorPtr& u, VectorPtr& v) { appendVectors(u, v); });
+
+  for (int8_t t = 0; t <= static_cast<int8_t>(TypeKind::INTERVAL_DAY_TIME);
+       t++) {
+    // VARCHAR and VARBINARY create the same C++ type, so we skip one of them.
+    if (static_cast<TypeKind>(t) != TypeKind::VARCHAR) {
+      VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH(
+          registerTypedVectors,
+          static_cast<TypeKind>(t),
+          m,
+          asModuleLocalDefinitions);
+    }
+  }
+
+  py::class_<DictionaryIndices>(
+      m, "DictionaryIndices", py::module_local(asModuleLocalDefinitions))
+      .def(
+          "__len__",
+          [](DictionaryIndices indices) {
+            return (indices.indices->size()) / sizeof(vector_size_t);
+          })
+      .def("__getitem__", [](DictionaryIndices indices, vector_size_t idx) {
+        checkBounds(indices, idx);
+        return indices.indices->as<vector_size_t>()[idx];
+      });
+
   m.def("from_list", [](const py::list& list) mutable {
     return pyListToVector(list, PyVeloxContext::getInstance().pool());
   });
+  m.def("constant_vector", [](const py::handle& obj, vector_size_t length) {
+    return pyToConstantVector(
+        obj, length, PyVeloxContext::getInstance().pool());
+  });
+  m.def("dictionary_vector", [](VectorPtr values, py::list& indices_list) {
```

Comment:
We also need to add null support to the dictionary . Need a utility that also takes nulls buffer.